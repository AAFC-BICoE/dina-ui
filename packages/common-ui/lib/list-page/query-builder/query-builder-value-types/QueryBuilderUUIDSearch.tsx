export interface TransformUUIDSearchToDSL {
  fieldPath: string;
  value: string;
}

/**
 * Using the query row for a UUID search, generate the elastic search request to be made.
 */
export function transformUUIDSearchToDSL({
  fieldPath,
  value
}: TransformUUIDSearchToDSL): any {
  return {
    term: {
      [fieldPath]: value
    }
  };
}
