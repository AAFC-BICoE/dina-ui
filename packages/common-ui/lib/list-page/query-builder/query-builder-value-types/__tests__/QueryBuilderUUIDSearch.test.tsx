import { transformUUIDSearchToDSL } from "../QueryBuilderUUIDSearch";

describe("QueryBuilderUUIDSearch", () => {
  describe("transformUUIDSearchToDSL function", () => {
    test("Field path and value provided", async () => {
      expect(
        transformUUIDSearchToDSL({
          fieldPath: "field_path_test",
          value: "value_test"
        })
      ).toMatchSnapshot();
    });
  });
});
