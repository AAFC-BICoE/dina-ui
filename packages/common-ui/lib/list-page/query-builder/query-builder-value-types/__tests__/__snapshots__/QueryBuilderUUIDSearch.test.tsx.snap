// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilderUUIDSearch transformUUIDSearchToDSL function Field path and value provided 1`] = `
Object {
  "term": Object {
    "field_path_test": "value_test",
  },
}
`;
